src/reducers/marksReducer.js

```diff
@@ -1,42 +1,31 @@
-import keys from '../data/key';
+import keys from "../data/key";
 
-const initToggleState = {
-  status: false
-};
-
-const initModalState = {
-  showModal: false
-};
-
-const initMarkerState = {
+const INITIAL_STATE = {
+  status: false,
+  showModal: false,
   marks: []
 };
 
-export const marksFetchReducer = (state = initMarkerState.marks, action) => {
-  switch (action.type) {
-    case keys.MARKS_FETCH_DATA_SUCCESS:
-      return action.marks;
-    case keys.MARK_ADD_DATA_SUCCESS:
-      return state.concat(action.marks);
-    default:
-      return state;
-  }
-};
-
-export const marksToggleReducer = (state = initToggleState.status, action) => {
-  switch (action.type) {
-    case keys.TOGGLE_MARKER:
-      return action.status;
-    default:
-      return state;
-  }
-};
-
-export const addMarkerReducer = (state = initModalState.showModal, action) => {
-  switch (action.type) {
-    case keys.ADD_MARKER:
-      return action;
-    default:
-      return state;
-  }
-};
+export const marksReducer = (state = INITIAL_STATE, action) => {
+    switch (action.type) {
+      case keys.MARKS_FETCH_DATA_SUCCESS:
+        return {
+          ...state,
+          marks: action.marks
+        }
+      case keys.MARK_ADD_DATA_SUCCESS:
+        return state.concat(action.marks);
+      case keys.TOGGLE_MARKER:
+        return {
+          ...state,
+          status: action.status
+        }        
+      case keys.ADD_MARKER:
+        return {
+          ...state,
+          showModal: action
```

Comment:
Here should be action.showModal since we are mapping it to props as "state.marksReducer.showModal" otherwise it will through error @abdiG